lib/berkshelf/locations/github.rb

```diff
@@ -1,7 +1,17 @@
 module Berkshelf
   class GithubLocation < GitLocation
+    HOST = 'github.com'
     def initialize(dependency, options = {})
-      options[:git] = "git://github.com/#{options.delete(:github)}.git"
+      protocol = Berkshelf::Config.instance.github.protocol || :https
+      case protocol
+      when :ssh
+        options[:git] = "git@#{HOST}:#{options.delete(:github)}.git"
+      when :https
+        options[:git] = "https://#{HOST}/#{options.delete(:github)}.git"
```

Comment:
the original reason this wasn't required, was I envisioned the berksfile format for specifying a git repo wouldn't change, and the yml would control which protocol was used.  I thought this would minimize the input on berksfile creators as well as consumers, thus the git protocol was still the default (due to reasons addressed in comments farther down), but consumers of berksfiles could globally override the protocol so they wouldn't have to essentially fork the berksfile's creator's repo.  Regardless however, I super excited to see this go in!